### PR TITLE
Implement initial Twitch widget

### PR DESCRIPTION
**Pull request by JamStop**
Looking for structure + naming feedback!
In particular, please look over how I did the link wrapping for the click-able portion of the widget, as well as the fonts file changes :)

Things I don't like that may not be within scope:
- Notification bell icon loads before the rest of the content, which can feel iffy
- Cached thumbnail of previous Youtube link might briefly be loaded in, which also feels iffy

https://www.loom.com/share/a2e96c94bd8f47ef8c2b2b73ea9c0cd0

### CODE REVIEW

File: src/fido/components/TwitchWidget.vue [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
+      <div>
+        <div :class="$style.Content">
+          <div :class="$style.Title" v-if="title_">{{ title_ }}</div>
+          <img :class="$style.Thumbnail" :src="stream_.thumbnail" v-if="stream_.thumbnail" />
+        </div>
+
+        <div :class="$style.FooterBar">
+          <img :class="$style.NotificationIcon" :src="images_.notificationIcon" v-if="!online_">
+          <div :class="$style.Label" v-if="label_">{{ label_ }}</div>
+        </div>
```
Comment (HunterLarco) [inline src/fido/components/TwitchWidget.vue]: Given that the label is dynamically sizer here we should be more explicit about text overflow:

```scss
.FooterBar {
  @include layout-horizontal;
}

.NotificationIcon {
  flex-shrink: 0;
}

.Label {
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
```

File: src/fido/sass/fonts.scss [hunk 2]
```diff
@@ -8,7 +8,27 @@
...
+  font-weight: 900;
 }
+
+@mixin fonts-twitch-tag {
+  @include _fonts-base;
+
+  font-family: HelveticaNeue-CondensedBold, Helvetica Neue, Helvetica;
+  font-size: 16px;
+  line-height: 18px;
+  font-weight: 900;
```
Comment (HunterLarco) [inline src/fido/sass/fonts.scss:33]: NIT: please alphabetically sort css properties

*likewise elsewhere*

File: src/fido/components/TwitchWidget.vue [hunk 3]
```diff
@@ -1,12 +1,39 @@
...
+    <a :href="url" target="blank" :class="$style.Link">
+      <div>
+        <div :class="$style.Content">
+          <div :class="$style.Title" v-if="title_">{{ title_ }}</div>
+          <img :class="$style.Thumbnail" :src="stream_.thumbnail" v-if="stream_.thumbnail" />
+        </div>
+
+        <div :class="$style.FooterBar">
+          <img :class="$style.NotificationIcon" :src="images_.notificationIcon" v-if="!online_">
+          <div :class="$style.Label" v-if="label_">{{ label_ }}</div>
```
Comment (HunterLarco) [inline src/fido/components/TwitchWidget.vue]: image

Note that the bell and the text don't have the same vertical alignment. We can fix this with some vertical margins on the notification icon and `vertical-align: middle`.

File: src/fido/components/TwitchWidget.vue [hunk 4]
```diff
@@ -1,12 +1,39 @@
...
+
+      <div :class="$style.TopBarRight">
+        <div :class="$style.Liveness_Online" v-if="stream_.game">
+          Live Now
+        </div>
+        <div :class="$style.Liveness_Offline" v-else>Offline</div>
+      </div>
+    </div>
+
+    <a :href="url" target="blank" :class="$style.Link">
```
[the remark this answers is not in the document]
Comment (JamStop) [reply src/fido/components/TwitchWidget.vue:17]: I tried it both way. This feels much better

File: src/fido/components/TwitchWidget.vue [hunk 5]
```diff
@@ -1,12 +1,43 @@
...
   <div :class="$style.Host">
-    <div>{{ JSON.stringify(user_, null, 2) }}</div>
-    <div>{{ JSON.stringify(stream_, null, 2) }}</div>
+    <div :class="$style.TopBar">
+      <div :class="$style.TopBarLeft">
+        <img :class="$style.Logo" :src="images_.logo" />
+      </div>
+
+      <div :class="$style.TopBarRight">
+        <div :class="$style.Liveness_Online" v-if="stream_.game">
```
Comment (HunterLarco) [inline src/fido/components/TwitchWidget.vue:9]: should this just be `online_`? if so, we can remove `stream_.game`

File: src/fido/components/TwitchWidget.vue [hunk 6]
```diff
@@ -30,13 +73,25 @@ export default {
         fetch(`${process.fido.flags.server}/api/v1/twitch?url=${this.url}`)
           .then((response) => response.json())
           .then(({ user, stream }) => {
-            this.user_ = user;
-            this.stream_ = stream;
+            if (!user) return;
+            this.user_.name = user.name;
```
Comment (HunterLarco) [inline src/fido/components/TwitchWidget.vue:77]: `user_.name` is unused

File: src/fido/components/TwitchWidget.vue [hunk 7]
```diff
@@ -46,15 +101,102 @@ export default {
...
+  padding: 5px 20px 0;
+}
+
+.Title {
+  @include fonts-twitch-title
+
+  color: #000;
+}
+
+.Thumbnail {
```
Comment (HunterLarco) [inline src/fido/components/TwitchWidget.vue:160]: `display: block`

File: src/fido/components/TwitchWidget.vue [hunk 8]
```diff
@@ -46,15 +101,102 @@ export default {
...
+}
+
+.Content {
+  padding: 5px 20px 0;
+}
+
+.Title {
+  @include fonts-twitch-title
+
+  color: #000;
```
Comment (HunterLarco) [inline src/fido/components/TwitchWidget.vue:157]: this should be unnecessary... default color is black

**Pull request merged**